be/src/storage/rowset/dictcode_column_iterator.h

```diff
@@ -25,134 +25,86 @@
 #include "runtime/global_dict/types.h"
 #include "simd/gather.h"
 #include "storage/range.h"
-#include "storage/rowset/column_iterator.h"
+#include "storage/rowset/column_iterator_wrapper.h"
 #include "storage/rowset/scalar_column_iterator.h"
 
 namespace starrocks {
 // DictCodeColumnIterator is a wrapper/proxy on another column iterator that will
 // transform the invoking of `next_batch(size_t*, Column*)` to the invoking of
 // `next_dict_codes(size_t*, Column*)`.
-class DictCodeColumnIterator final : public ColumnIterator {
+class DictCodeColumnIterator final : public ColumnIteratorWrapper {
 public:
     using Column = starrocks::Column;
     using ColumnPredicate = starrocks::ColumnPredicate;
     // does not take the ownership of |iter|.
-    DictCodeColumnIterator(ColumnId cid, ColumnIterator* iter) : _cid(cid), _col_iter(iter) {}
+    explicit DictCodeColumnIterator(ColumnId cid, ColumnIterator* iter)
+            : ColumnIteratorWrapper(iter, kDontTakeOwnership), _cid(cid) {}
 
     ~DictCodeColumnIterator() override = default;
 
     ColumnId column_id() const { return _cid; }
 
-    ColumnIterator* column_iterator() const { return _col_iter; }
+    ColumnIterator* column_iterator() const { return _iter; }
 
-    [[nodiscard]] Status next_batch(size_t* n, Column* dst) override { return _col_iter->next_dict_codes(n, dst); }
+    Status next_batch(size_t* n, Column* dst) override { return _iter->next_dict_codes(n, dst); }
 
-    [[nodiscard]] Status next_batch(const SparseRange<>& range, Column* dst) override {
-        return _col_iter->next_dict_codes(range, dst);
-    }
-
-    [[nodiscard]] Status fetch_values_by_rowid(const rowid_t* rowids, size_t size, Column* values) override {
-        return _col_iter->fetch_values_by_rowid(rowids, size, values);
-    }
-
-    [[nodiscard]] Status seek_to_first() override { return _col_iter->seek_to_first(); }
-
-    [[nodiscard]] Status seek_to_ordinal(ordinal_t ord) override { return _col_iter->seek_to_ordinal(ord); }
-
-    ordinal_t get_current_ordinal() const override { return _col_iter->get_current_ordinal(); }
-
-    bool all_page_dict_encoded() const override { return _col_iter->all_page_dict_encoded(); }
-
-    int dict_lookup(const Slice& word) override { return _col_iter->dict_lookup(word); }
-
-    [[nodiscard]] Status next_dict_codes(size_t* n, Column* dst) override { return _col_iter->next_dict_codes(n, dst); }
-
-    [[nodiscard]] Status decode_dict_codes(const int32_t* codes, size_t size, Column* words) override {
-        return _col_iter->decode_dict_codes(codes, size, words);
-    }
-
-    [[nodiscard]] Status get_row_ranges_by_zone_map(const std::vector<const ColumnPredicate*>& predicates,
-                                                    const ColumnPredicate* del_predicate,
-                                                    SparseRange<>* row_ranges) override {
-        return _col_iter->get_row_ranges_by_zone_map(predicates, del_predicate, row_ranges);
-    }
```

Comment:
Replaced the implementations here with the default implementations in ColumnIteratorWrapper.